boardgame-ui/src/pages/HomePage.vue

```diff
@@ -1,3 +1,129 @@
+      <div class="hero-body">
+        <div class="container">
+          <h1 class="title">
+            {{$t('home.hello')}} {{ currentUser ? currentUser.name : "" }} !
+            <span class="icon">
+              <i class="hero-smile fa fa-smile" />
+            </span>
+          </h1>
+          <h2 class="subtitle">
+            {{$t('home.welcome')}} {{$t('app.appName')}}.
+          </h2>
+        </div>
+      </div>
+    </section>
+
+    <div class="container">
+      <b-loading :is-full-page="false" :active.sync="isFetching" />
+      <section class="section" v-if="ongoingEvent">
+        <router-link tag="button" class="button is-large is-fullwidth" :to="{name: 'event', params: {eventid: ongoingEvent.id}}">
+          <p>{{ongoingEvent.name}}</p> <span class="icon"><i class="fa fa-arrow-right"/></span>
+        </router-link>
+      </section>
+
+      <section class="section" v-if=userStats>
+        <div class="columns">
+          <div class="column has-text-centered">
+            <p class="heading">{{$t('home.stats.owned')}}</p>
+            <p class="title">{{userStats.owned}}</p>
+          </div>
+
+          <div class="column has-text-centered">
+            <p class="heading">{{$t('home.stats.played')}}</p>
+            <p class="title">{{userStats.played}}</p>
+          </div>
+
+          <div class="column has-text-centered">
+            <p class="heading">{{$t('home.stats.attended')}}</p>
+            <p class="title">{{userStats.attended}}</p>
+          </div>
+
+          <div class="column has-text-centered" v-if="userStats.most_played.count > 0">
+            <p class="heading">{{$t('home.stats.most_played')}}</p>
+            <p class="title">{{userStats.most_played.board_game.name}}</p>
+            <p>({{userStats.most_played.count}} {{$t('home.stats.times')}})</p>
+          </div>
+
+          <div class="column has-text-centered" v-if="userStats.most_played.count > 0">
+            <p class="heading">{{$t('home.stats.time_played')}}</p>
+            <p class="title">{{userStats.play_time}} {{$t('home.stats.minutes')}}</p>
+          </div>
+        </div>
+      </section>
+
+      <section class="section" v-if="userActivities">
+        <h2 class="subtitle">{{$t('home.recent-activities')}}</h2>
+        <ActivityBox v-for="index in userActivities.length" :key=index :activity=userActivities[index-1] ></ActivityBox>
+      </section>
+    </div>
+  </div>
+
 </template>
+
+<script>
+import Event from '@/utils/api/Event';
+import User from '@/utils/api/User';
+import ActivityBox from '@/components/activities/ActivityBox';
+
+export default {
+  name: 'HomePage',
+  data() {
+    return {
+      userStats: null,
+      userActivities: null,
+      ongoingEvent: null,
+      isFetching: false
+    };
+  },
+  components: {
+    ActivityBox
+  },
+  computed: {
+    currentUser() {
+      return this.$store.state.currentUser;
+    }
+  },
+  async created() {
+    if (!this.currentUser) {
```

Comment:
As we discussed, if this line is executed before the store is initialized, the data is not fetched, and the page is never populated.
Several solutions to avoid this issue:
1. Create a new component AuthenticatedHomePage, include in it all data and template related to current user (line 19 to 62), display it only if there is a current user (v-if="currentUser") and fetch data in the created hook of this new component
2. In App.vue, prevent the creation of <router-view> component until the store has been initialized
3. Create a watcher for currentUser, and reload data if its value is changed

I think the second is the easiest, but I let you decide, and don't hesitate to ask if you need help! :-)

Reply:
Actually, there would be very few common content between the authenticated home page and the other. I think it would be better to have two different pages. I was not sure how to update the router (the change of "home" to "login" redirect was a temporary workaround in this direction), especially because the path would be the same (i.e. '/'). I was not sure if it was possible to have several pages with the same path.

Reply:
That's why I think it would make sense to have one component responsible for the home page of the guest (let us call it GuestHomePage), and one responsible for the home page of an authenticated user (UserHomePage).
Then, the template of HomePage.vue would simply be:
```
<template>
  <user-home-page v-if="currentUser"></user-home-page>
  <guest-home-page v-else></guest-home-page>
</template>
```
I don't know if there is another way to handle this in vue-router, I can investigate :-)

Reply:
Let's go for this solution then 👍